pkgs/top-level/all-packages.nix

```diff
@@ -6203,6 +6201,10 @@ with pkgs;
   volumeicon = callPackage ../tools/audio/volumeicon { };
 
   waf = callPackage ../development/tools/build-managers/waf { python = python3; };
+  wafHook = makeSetupHook {
+    deps = [ python ];
+    substitutions = { inherit waf; };
+  } ../development/tools/build-managers/waf/setup-hook.sh;
```

Comment:
should wafHook live in pythonPackages ? or provide waf2Hook for python2 ? Maybe it's ok if you had no problem so far but I know at least one waf-based program (outside nixpkgs) that only supports python2

Reply:
waf is usually vendored so I would think it would always support python2? Not 100% sure but I had no problems doing it this way here.